js_modules/dagit/packages/core/src/app/analytics.tsx

```diff
@@ -0,0 +1,51 @@
+import * as React from 'react';
+import {useLocation, useRouteMatch} from 'react-router-dom';
+
+export interface GenericAnalytics {
+  identify?: (userId: string) => void;
+  page: (path: string, specificPath: string) => void;
+  track: (eventName: string, properties?: Record<string, any>) => void;
+}
+
+export const AnalyticsContext = React.createContext<GenericAnalytics | null>(null);
+
+const PAGEVIEW_DELAY = 300;
+
+const usePageContext = () => {
+  const match = useRouteMatch();
+  const {pathname: specificPath} = useLocation();
+  const {path} = match;
+  return React.useMemo(() => ({path, specificPath}), [path, specificPath]);
+};
+
+export const useTrackPageView = () => {
+  const analytics = React.useContext(AnalyticsContext);
+  const {path, specificPath} = usePageContext();
+
+  React.useEffect(() => {
+    let timer: ReturnType<typeof setTimeout>;
+
+    if (analytics) {
+      // Wait briefly to allow redirects.
+      timer = setTimeout(() => {
+        analytics.page(path, specificPath);
+      }, PAGEVIEW_DELAY);
+    }
```

Comment:
Should we add some type of warning/error logging if `analytics` is not defined? ie. They're somehow using `useTrackPageView` outside of a tree with `AnalyticsContext.Provider` wrapped around it? Ie. Its unlikely to happen but would be nice to have